A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,405 @@
+## Title
+
+*   Author(s): Yash Tibrewal (@yashykt)
+*   Approver: Mark Roth (@markdroth)
+*   Status: In Review
+*   Implemented in: <language, ...>
+*   Last updated: Feb 23, 2024
+*   Discussion at: https://groups.google.com/g/grpc-io/c/VfZOj7940Dc
+
+## Abstract
+
+Describe a cross-language architecture for collecting non-per-call metrics.
+
+## Background
+
+[A66: OpenTelemetry Metrics](A66-otel-stats.md) is a spec for gRPC to collect
+client/server per-attempt/call metrics for OpenTelemetry. It details an
+architecture that uses a CallTracer approach to record various per-call metrics
+like latency and message sizes. This approach works for per-attempt/call metrics
+but does not work so well for non-per-call metrics such as xDS metrics, load
+balancing policy metrics, resolver metrics and transport metrics which are not
+necessarily tied to a particular call. This calls for a newer architecture that
+allows to collect such metrics.
+
+### Related Proposals:
+
+*   [A66: OpenTelemetry Metrics](A66-otel-stats.md)
+*   [A78: gRPC OTel Metrics for WRR, Pick First and XdsClient](https://github.com/grpc/proposal/pull/419)
+
+## Proposal
+
+### Overall Goals / Requirements
+
+*   Support non-per-call metrics -
+    [A78: gRPC OTel Metrics for WRR, Pick First and XdsClient](https://github.com/grpc/proposal/pull/419)
+    defines metrics for WRR, Pick-First and XdsClient components. In the future,
+    we would want to record metrics for other components as well. Additionally,
+    third-party plugins to gRPC (custom load balancing policies/resolvers) might
+    want to use the same architecture to record metrics as well.
+*   Channel-scoped stats plugins - Applications should be able to register stats
+    plugins such that metrics for only certain channels are recorded on those
+    plugins. (Current implementations only support global registration of stats
+    plugins.) There can be multiple stats plugins registered as well.
+*   gRPC/Third-party plugins control the metric instrument definition - As gRPC
```

Comment:
We should also explicitly state somewhere that the "grpc." namespace is reserved for gRPC, and metrics provided by third-party plugins should use their own namespaces.